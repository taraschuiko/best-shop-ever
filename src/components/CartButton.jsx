import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import numberOfProductsInCartSelector from "../selectors/numberOfProductsInCart";
import totalPriceSelector from "../selectors/totalPrice";

export function CartButton({ productsQuantity, totalPrice }) {
  return (
    <Link to="/cart" className="cart-button col-12">
      You have {productsQuantity} products in your cart. The total price is
      {" $"}
      {totalPrice}
    </Link>
  );
}

const mapState = state => ({
  productsQuantity: numberOfProductsInCartSelector(state.cart),
  totalPrice: totalPriceSelector(state.cart)
});

export default connect(mapState)(CartButton);
