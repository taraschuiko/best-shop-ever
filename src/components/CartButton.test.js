import React from "react";
import renderer from "react-test-renderer";
import { CartButton } from "./CartButton";
import { BrowserRouter as Router } from "react-router-dom";
import { Link } from "react-router-dom";

describe("CartButton", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <CartButton productsQuantity={5} totalPrice={"100.10"} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has correct link", () => {
    const testInstance = renderer.create(
      <Router>
        <CartButton productsQuantity={5} totalPrice={"100.10"} />
      </Router>
    ).root;
    expect(testInstance.findByType(Link).props.to).toBe("/cart");
  });
});
