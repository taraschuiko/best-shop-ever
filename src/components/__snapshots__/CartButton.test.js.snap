// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartButton renders correctly 1`] = `
<a
  className="cart-button col-12"
  href="/cart"
  onClick={[Function]}
>
  You have 
  5
   products in your cart. The total price is
   $
  100.10
</a>
`;
